functions/Handlers/emailHandler.js

```diff
@@ -0,0 +1,109 @@
+const Functions = require('firebase-functions/lib/index');
+const NodeMailer = require("nodemailer");
+const Util = require("../util.js");
+
+const env = Functions.config();
+const gmail = NodeMailer.createTransport({
+    service: "gmail",
+    pool: true,
+    auth: {
+        user: env.gmail.email,
+        pass: env.gmail.password
+        //user: env.myGmail.email,
+        //pass: env.myGmail.password
+    }
+});
+
+const sendEmail = function (info) {
+    return new Promise((res, rej) => {
+        if (!info.to) {
+            rej("No receivers were specified.");
+        }
+
+        let messageOptions = Object.assign({}, info);
+        messageOptions.from = `musicmattersbookings@gmail.com <"Music Matters Bookings">`;
+
+        gmail.sendMail(messageOptions).then(res).catch(rej);
+    });
+};
+exports.sendEmail = sendEmail;
+
+exports.sendArtistConfirmation = function (client, event, venue, acPDF) {
+    let emailSubject = venue.name + " - Artist Confirmation";
+    let emailBody = "Hello!\n\n" +
+        "Attached is your confirmation for your performance" + " at " + venue.name + " at " + Util.toAMPM(event.start) + "\n" +
+        "Please don't hesitate to reply back to this email if you have any questions.";
+
+    return sendEmail({
+        to: client.email,
+        subject: emailSubject,
+        text: emailBody,
+        attachments: [
+            {
+                filename: "Confirmation " + event.date + ".pdf",
+                content: acPDF
+            }
+        ]
+    });
+};
+
+exports.sendInvoice = function (client, event, venue, invoicePDF) {
+    let emailSubject = client.stage + " - Invoice - " + Util.toUS(event.date);
+    let emailBody = "To whom it may concern,\n\n" +
+        "Attached is the invoice for " + client.stage + ", who is performing at your venue on " + Util.toUSText(event.date) + ". " +
```

Comment:
I know this may seem redundant, but perhaps "your venue" should be replaced with "venue.name" to explicitly list the name of the venue in the email?
I don't believe this is a problem now, but if Mr. Moody ever does business with a group that manages multiple venues, it would be more convenient to list the venue in the email itself rather than just in the attached document.
Same goes for the sendCalendar and sendBookingList functions below.

Reply:
That's a good idea, I'll add that in before merging.